packages/test/src/workflows/fill-memory.ts

```diff
@@ -0,0 +1,19 @@
+export async function fillMemory(): Promise<number> {
+  // It looks like JIT compilation of the following code affects the way the out of memory error
+  // will get reported. That is, once the code has been optimized, the out of memory condition is
+  // very likely to trigger a low level (native-like) error, which may terminate the whole process,
+  // rather than just the Worker Thread.
+  //
+  // Wrapping code in an eval block delays JIT compilation, thus reducing the probability of flakes.
+
+  // Allocate 32GB of memory
```

Comment:
I removed that comment, it is misleading. How much we allocate is not really important.

There are different ways that Node may fail due to an "out of memory" condition, and the criterias for each of them are totally undocumented. This test is about triggering a very specific type of OOM, the `ERR_WORKER_OUT_OF_MEMORY` error, which seems to be related to specific heap spaces (probably the young gen) and/or to JIT compilation.

This code has been resulting in a ERR_WORKER_OUT_OF_MEMORY error in all setups I have tried; that includes all platforms, for Node 16, 18 and 20, with various quantity of RAM, more than 95% of the time. The error happens almost immediately, way before reaching the order of GBs. In fact, if I add a `console.log`, it appears that the process crashes on the first execution of the loop!

Yet, if I remove the loop, or reduce the higher limit on the loop (threshold seems to be at ~64), then it no longer fails, but execution becomes much slower. Similarly, if I change the code to allocate larger arrays, then it tends to fail early by throwing an error (rather than killing the process). And if I add an `await wf.sleep();` in the loop, then it can allocate much more memory without crashing.

All of this suggests interractions of V8's JIT compiler and/or GC memory allocator. The higher limit on the loop probably triggers an early JIT optimization (which would not be deemed useful if the loop is expected to run less than a certain number of time). Similarly, when allocating larger arrays, it's possible that the heap allocator decides to allocate directly from the perm gen space (which would have a significantly higher capacity) rather than from the young gen (which has a default capacity of 16 MB according to Node's documentation).

Now, given all of those uncertainties, this test is likely to be fragile. Why then do I care that much about reproducing this very specific error pattern? For some unknown reason, that's an error pattern that users are observing regularly in practice, while I have seen no report of other types of OOM errors in a long time (except for docker/pods memory limit misconfiguration, but there's nothing more that the SDK can do about those).